include/revng/Pipeline/Global.h

```diff
@@ -149,6 +166,25 @@ class TupleTreeGlobal : public Global {
 
   const TupleTree<Object> &get() const { return Value; }
   TupleTree<Object> &get() { return Value; }
+
+  PathSerializationPrimitives getSerializationPrimitives() const override {
+    return PathSerializationPrimitives::make<Object>();
+  }
+
+  void collectReadFields(const TargetInStep &Target,
+                         PathTargetBimap &Out) override {
+    const auto &AsConst = Value;
+    auto Results = revng::Tracking::collect(AsConst);
+    for (auto &Result : Results.ExactVectors)
+      Out.insert(Target, Result);
+    for (auto &Result : Results.ExactVectors)
+      Out.insert(Target, Result);
```

Comment:
1: ok
2: yep, it was a bug
3: we do not need to differentiate between them now, but we will need to differentiate between them when we introduce the hash discriminator when doing the cold start, because exact vectors will need to contribute to the hash in a different way.

Reply:
3. Leave a comment about this.

Reply:
i put the comment on the struct instead of that snippet.